src/helpers/MiscFunctions.cpp

```diff
@@ -503,6 +503,45 @@ int getWorkspaceIDFromString(const std::string& in, std::string& outName) {
     return result;
 }
 
+std::string cleanCmdForWorkspace(const std::string& inWorkspaceName, std::string dirtyCmd) {
+
+
+    std::string cmd = removeBeginEndSpacesTabs(dirtyCmd);
+
+    if (!cmd.empty()) {
+        std::string       rules;
+        const std::string workspaceRule = "workspace " + inWorkspaceName;
+
+        if (cmd[0] == '[') {
+            const int closingBracketIdx = cmd.find_last_of(']');
+            auto      tmpRules          = cmd.substr(1, closingBracketIdx - 1);
+            cmd                         = cmd.substr(closingBracketIdx + 1);
+
+            auto rulesList = CVarList(tmpRules, 0, ';');
+
+            bool hadWorkspaceRule = false;
+            rulesList.map([&](std::string& rule) {
+                if (rule.find("workspace") == 0) {
+                    rule             = workspaceRule;
+                    hadWorkspaceRule = true;
+                }
+            });
+
+            if (!hadWorkspaceRule) {
```

Comment:
style: no {} around short ifs

Reply:
I will disregard this. Short ifs are single-line. What you showed is not. I wish clang-format would support that. Maybe I should write a hyprformat.

Reply:
```yml
# clang-format
RemoveBracesLLVM: true
```

Even though it removes braces from for and while, it is still effective.